app/models/responsible_body.rb

```diff
@@ -20,6 +24,17 @@ class ResponsibleBody < ApplicationRecord
 
   after_update :maybe_generate_user_changes
 
+  def calculate_virtual_caps!
+    virtual_cap_pools.each(&:recalculate_caps!)
+  end
+
+  def add_school_to_virtual_cap_pools!(school)
```

Comment:
is there a safeguard to accidentally adding a school twice to the pool or is this overkill?